### ISSUE TITLE
XslLoadException: 'xsl:import' element children must precede all other children of the 'xsl:transform' element.

### ISSUE THREAD

**Issue by btol** [schxslt/schxslt#317]
I am trying to validate a message against a Schematron schema using C#. I thought this was going to be as simple as:

Step 1. Transform the schema using _pipeline-for-svrl.xsl_
Step 2. Transform the message using the result from step 1.
Step 3. Extract any error messages in the resulting document from step 2.

However when i try to load the _pipeline-for-svrl.xsl_ transform I get the following error message:
`'xsl:import' element children must precede all other children of the 'xsl:transform' element.`

Am I missing something?

The problem seems to be in pipeline.xsl.

The serialized exception:
```
System.Xml.Xsl.XslLoadException: 'xsl:import' element children must precede all other children of the 'xsl:transform' element. An error occurred at C:\Project\TracsFlow\Repo\Current\Source\Labs\Beast4\FlowBEAst\TransformRepro\bin\Debug\net8.0\Transform\2.0\pipeline.xsl, (8, 4).
   at System.Xml.Xsl.XslCompiledTransform.LoadInternal(Object stylesheet, XsltSettings settings, XmlResolver stylesheetResolver, XmlResolver originalStylesheetResolver)
   at System.Xml.Xsl.XslCompiledTransform.Load(String stylesheetUri, XsltSettings settings, XmlResolver stylesheetResolver)
   at Program.<Main>$(String[] args) in C:\Project\TracsFlow\Repo\Current\Source\Labs\Beast4\FlowBEAst\TransformRepro\Program.cs:line 21
```
The error occurs when executing the following statement:
`xslt.Load(compileForSvrlPath, settings, resolver);`

The file Program.cs below is the entire code for the c# console test program.

```
using System.Xml.Xsl;
using System.Xml;

var assemblyPath = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location)!;
var compileForSvrlPath = Path.Combine(assemblyPath, "Transform\\2.0\\pipeline-for-svrl.xsl");

var xslt = new XslCompiledTransform();

var settings = new XsltSettings
{
    EnableDocumentFunction = true,
    EnableScript = true
};

var resolver = new XmlUrlResolver
{
    Credentials = System.Net.CredentialCache.DefaultCredentials
};


try
{
    xslt.Load(compileForSvrlPath, settings, resolver);

    var xmlDoc = new XmlDocument();
    var schema = Path.Combine(assemblyPath, "PEPPOLBIS-T120.sch");
    xmlDoc.Load(schema);

    using var transformResult = new MemoryStream();

    using var stringWriter = new StringWriter();
    xslt.Transform(xmlDoc, null, stringWriter);
    Console.WriteLine("Resulting transform:");
    Console.WriteLine(stringWriter.ToString());

    // ToDo: Use the resulting transform to validate the example document
}
catch (Exception ex)
{
    Console.WriteLine(ex.ToString());
}
```

The project file is as follows:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
	  <None Update="transform\**\*.*">
		  <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
	  </None>
  </ItemGroup>

</Project>
```

**Comment by dmj** [comment]
Thanks for reporting! This is a subtle bug in SchXslt: It is true that xsl:import element must precede all other element children of an xsl:stylesheet element, including any xsl:include element (See: https://www.w3.org/TR/xslt20/#import). Will fix this in a bugfix-release v1.9.6

**Comment by btol** [comment]
Thank You for the quick response!
However, I now realize that .NET (8) only has built in support for XSLT 1.0 while Schematron requires XSLT 2.0.
My understanding is that this would require a commercial license for SaxonCS from Saxonica.
